packages/react-instantsearch/src/connectors/connectInfiniteHits.js

```diff
@@ -66,7 +66,7 @@ export default createConnector({
     const id = getId();
     const nextPage = searchState[id] ?
       Number(searchState[id]) + 1 :
-      1;
+      0;
```

Comment:
I'm agree, this is a monkey patch, but looks a problem related with what index is considered as first index